File: frontend/ai-control-center-angular/src/app/features/directions/direction.models.ts
export type DirectionStatus = 'Active' | 'Inactive';

export interface DirectionDto {
  id: string;
  name: string;
  code: string;
  description: string | null;
  icon: string | null;
  status: DirectionStatus;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
  archivedAt: string | null;
  version: number;
  isArchived: boolean;
}

export interface DirectionListDto {
  items: DirectionDto[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}
export interface DirectionContent {
  name: string;
  code: string;
  description: string | null;
  icon: string | null;
}
export interface CreateDirectionRequest extends DirectionContent {
  status: DirectionStatus;
  sortOrder: number;
}
export interface UpdateDirectionRequest extends DirectionContent {
  status: DirectionStatus;
  sortOrder: number;
  version: number;
}
export interface DirectionFilters {
  includeArchived: boolean;
  status?: DirectionStatus;
  search?: string;
  page: number;
  pageSize: number;
}

File: src/Services/ControlPlane/AiControlCenter.ControlPlane.Application/DirectionModels.cs
using AiControlCenter.ControlPlane.Domain;

namespace AiControlCenter.ControlPlane.Application;

public sealed record ListDirectionsQuery(
    bool IncludeArchived = false,
    DirectionStatus? Status = null,
    string? Search = null,
    int Page = 1,
    int PageSize = 20)
{
    public const int DefaultPageSize = 20;
    public const int MaximumPageSize = 100;
}

public sealed record GetDirectionQuery(Guid Id);

public sealed record CreateDirectionCommand(
    string Name,
    string Code,
    string? Description,
    string? Icon,
    DirectionStatus Status,
    int SortOrder);

public sealed record UpdateDirectionCommand(
    string Name,
    string Code,
    string? Description,
    string? Icon,
    DirectionStatus Status,
    int SortOrder,
    uint Version);

public sealed record ChangeDirectionStatusCommand(DirectionStatus Status, uint Version);

public sealed record ChangeDirectionSortOrderCommand(int SortOrder, uint Version);

public sealed record ArchiveDirectionCommand(uint Version);

public sealed record RestoreDirectionCommand(uint Version);

public sealed record DirectionDto(
    Guid Id,
    string Name,
    string Code,
    string? Description,
    string? Icon,
    DirectionStatus Status,
    int SortOrder,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt,
    DateTimeOffset? ArchivedAt,
    uint Version)
{
    public bool IsArchived => ArchivedAt is not null;
}

public sealed record DirectionListDto(
    IReadOnlyCollection<DirectionDto> Items,
    int Page,
    int PageSize,
    int TotalCount)
{
    public int TotalPages => TotalCount == 0 ? 0 : (int)Math.Ceiling((double)TotalCount / PageSize);
}
